### PR TITLE
MINOR: fix typos for client

### PR DESCRIPTION
**Pull request by berg223**
*More detailed description of your change,
Fix typos for client*

### Committer Checklist (excluded from commit message)
- [x] Verify design and implementation 
- [x] Verify test coverage and CI build status
- [x] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by berg223** [comment]
test this please

File: clients/src/main/java/org/apache/kafka/clients/consumer/internals/OffsetFetcher.java [hunk 1]
```diff
@@ -430,7 +430,7 @@ public void onFailure(RuntimeException e) {
      * leaders available. Topic partitions from `timestampsToSearch` that do not have their leader
      * available are added to `partitionsToRetry`
      *
-     * @param timestampsToSearch The mapping from partitions ot the target timestamps
+     * @param timestampsToSearch The mapping from partitions of the target timestamps
```
Comment (showuon) [inline clients/src/main/java/org/apache/kafka/clients/consumer/internals/OffsetFetcher.java]: I think this is `to`

**Review by showuon — COMMENTED** [review]
@xiaocairush , thanks for the patch. Really impressive you found all these typos. Left a comment.

File: clients/src/main/java/org/apache/kafka/clients/consumer/internals/OffsetFetcher.java (hunk 1 above)
Comment (berg223) [reply clients/src/main/java/org/apache/kafka/clients/consumer/internals/OffsetFetcher.java]: Corrected to **"to"**

**Review by kirktrue — APPROVED** [review]
LGTM 👍

Thanks for fighting the good fight 😆

File: clients/src/test/java/org/apache/kafka/common/utils/UtilsTest.java [hunk 2]
```diff
@@ -854,7 +854,7 @@ public void shouldThrowOnInvalidDateFormatOrNullTimestamp() {
 
     private void checkExceptionForGetDateTimeMethod(Executable executable) {
         assertTrue(assertThrows(ParseException.class, executable)
-            .getMessage().contains("Unparseable date"));
+            .getMessage().contains("Unparsable date"));
```
Comment (divijvaidya) [inline clients/src/test/java/org/apache/kafka/common/utils/UtilsTest.java]: we have some test failures associated with this: https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-13884/3/testReport/junit/org.apache.kafka.common.utils/UtilsTest/Build___JDK_11_and_Scala_2_13___shouldThrowOnInvalidDateFormatOrNullTimestamp__/ 

Please fix the source code along with the test validation.

**Review by divijvaidya — CHANGES_REQUESTED** [review]

File: clients/src/test/java/org/apache/kafka/common/utils/UtilsTest.java (hunk 2 above)
Comment (berg223) [reply clients/src/test/java/org/apache/kafka/common/utils/UtilsTest.java]: fixed!

**Review by divijvaidya — APPROVED** [review]
Thank you for the change. It looks good now.

Test failures are unrelated to this change. 
```
[Build / JDK 8 and Scala 2.12 / org.apache.kafka.clients.consumer.KafkaConsumerTest.testSecondPollWithDeserializationErrorThrowsRecordDeserializationException()](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-13884/4/testReport/junit/org.apache.kafka.clients.consumer/KafkaConsumerTest/Build___JDK_8_and_Scala_2_12___testSecondPollWithDeserializationErrorThrowsRecordDeserializationException__/)
[Build / JDK 8 and Scala 2.12 / org.apache.kafka.connect.mirror.integration.IdentityReplicationIntegrationTest.testReplicateSourceDefault()](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-13884/4/testReport/junit/org.apache.kafka.connect.mirror.integration/IdentityReplicationIntegrationTest/Build___JDK_8_and_Scala_2_12___testReplicateSourceDefault__/)
[Build / JDK 8 and Scala 2.12 / org.apache.kafka.connect.mirror.integration.MirrorConnectorsIntegrationExactlyOnceTest.testOffsetTranslationBehindReplicationFlow()](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-13884/4/testReport/junit/org.apache.kafka.connect.mirror.integration/MirrorConnectorsIntegrationExactlyOnceTest/Build___JDK_8_and_Scala_2_12___testOffsetTranslationBehindReplicationFlow__/)
[Build / JDK 8 and Scala 2.12 / kafka.zk.ZkMigrationIntegrationTest.[1] Type=ZK, Name=testNewAndChangedTopicsInDualWrite, MetadataVersion=3.4-IV0, Security=PLAINTEXT](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-13884/4/testReport/junit/kafka.zk/ZkMigrationIntegrationTest/Build___JDK_8_and_Scala_2_12____1__Type_ZK__Name_testNewAndChangedTopicsInDualWrite__MetadataVersion_3_4_IV0__Security_PLAINTEXT/)
[Build / JDK 17 and Scala 2.13 / org.apache.kafka.common.network.SslTransportLayerTest.[2] tlsProtocol=TLSv1.2, useInlinePem=true](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-13884/4/testReport/junit/org.apache.kafka.common.network/SslTransportLayerTest/Build___JDK_17_and_Scala_2_13____2__tlsProtocol_TLSv1_2__useInlinePem_true/)
[Build / JDK 17 and Scala 2.13 / org.apache.kafka.connect.mirror.integration.MirrorConnectorsIntegrationExactlyOnceTest.testOffsetTranslationBehindReplicationFlow()](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-13884/4/testReport/junit/org.apache.kafka.connect.mirror.integration/MirrorConnectorsIntegrationExactlyOnceTest/Build___JDK_17_and_Scala_2_13___testOffsetTranslationBehindReplicationFlow__/)
[Build / JDK 17 and Scala 2.13 / integration.kafka.server.FetchFromFollowerIntegrationTest.testRackAwareRangeAssignor()](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-13884/4/testReport/junit/integration.kafka.server/FetchFromFollowerIntegrationTest/Build___JDK_17_and_Scala_2_13___testRackAwareRangeAssignor__/)
[Build / JDK 17 and Scala 2.13 / kafka.api.TransactionsTest.testBumpTransactionalEpoch(String).quorum=kraft](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-13884/4/testReport/junit/kafka.api/TransactionsTest/Build___JDK_17_and_Scala_2_13___testBumpTransactionalEpoch_String__quorum_kraft/)
[Build / JDK 17 and Scala 2.13 / org.apache.kafka.controller.QuorumControllerTest.testBalancePartitionLeaders()](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-13884/4/testReport/junit/org.apache.kafka.controller/QuorumControllerTest/Build___JDK_17_and_Scala_2_13___testBalancePartitionLeaders__/)
```

**Pull request merged**